Sources/CMakeLists.txt

```
@@ -12,12 +12,28 @@
 ##
 ##===----------------------------------------------------------------------===##
 
-add_subdirectory(_FoundationCShims)
-
-# Disable the macro build on Windows until we can correctly build it for the host architecture
-if(NOT CMAKE_SYSTEM_NAME STREQUAL Windows)
-	add_subdirectory(FoundationMacros)
+include(ExternalProject)
+if(CMAKE_HOST_WIN32)
+  set(_FoundationMacrosSwiftFlags "-use-ld=lld")
+endif()
+ExternalProject_Add(FoundationMacros
+  SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/FoundationMacros"
+  PREFIX "${CMAKE_BINARY_DIR}/_deps"
+  BINARY_DIR "macros"
+  CMAKE_ARGS
+    -DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}
+    -DCMAKE_SYSTEM_NAME=${CMAKE_HOST_SYSTEM_NAME}
```

```suggestion
    -DCMAKE_SYSTEM_NAME=${CMAKE_HOST_SYSTEM_NAME}
    -DCMAKE_Swift_COMPILER=${CMAKE_Swift_COMPILER}
```
To ensure using a consistent compiler between libraries and macros, I think it might be better to propagate `CMAKE_Swift_COMPILER` instead of assuming swiftc is available in PATH.

---

This should no longer be a concern given that we do not do the internal build.